File: Baekjoon/문제풀이/괄호/yuri.java
import java.util.*;

public class BracketCheck {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        sc.nextLine(); // 버퍼 비우기
        
        for (int i = 0; i < t; i++) {
            String brackets = sc.nextLine();
            Stack<Character> stack = new Stack<>();
            boolean isValid = true;
            
            for (int j = 0; j < brackets.length(); j++) {
                char c = brackets.charAt(j);
                if (c == '(') {
                    stack.push(c);
                } else if (c == ')') {
                    if (!stack.isEmpty()) {
                        stack.pop();
                    } else {
                        isValid = false;
                        break;
                    }
                }
            }
            
            if (isValid && stack.isEmpty()) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
    }
}

File: Baekjoon/문제풀이/괄호/yuri.py
t = int(input())

for _ in range(t):
    stack = []
    is_valid = True
    brackets = input()
    
    for char in brackets:
        if char == "(":
            stack.append(char)
        elif char == ")":
            if stack:
                stack.pop()
            else:
                is_valid = False
                break
    
    if is_valid and not stack:
        print("YES")
    else:
        print("NO")
